### PR TITLE
New REST endpoints - Spring course

### PR DESCRIPTION
**Pull request by sniochup**

### CODE REVIEW

File: src/main/java/pl/akai/bookcrossing/book/BookRestController.java [hunk 1]
```diff
@@ -6,12 +6,31 @@
...
+import java.util.List;
 
 @RestController
 @RequiredArgsConstructor
 public class BookRestController {
 
     private final BookBean bookBean;
+    private final OpinionBean opinionBean;
+
+    @GetMapping("/books/list")
```
Comment (Szczepaniak-M) [inline src/main/java/pl/akai/bookcrossing/book/BookRestController.java]:
```suggestion
    @GetMapping("/api/books")
```

File: src/main/java/pl/akai/bookcrossing/book/BookRestController.java [hunk 2]
```diff
@@ -6,12 +6,31 @@
...
+    public List<Book> booksList() {
+        return bookBean.getAllBooks();
+    }
+
+    @GetMapping("/book/details/{id}")
```
Comment (Szczepaniak-M) [inline src/main/java/pl/akai/bookcrossing/book/BookRestController.java]:
```suggestion
    @GetMapping("/api/books/{id}")
```

File: src/main/java/pl/akai/bookcrossing/book/BookRestController.java [hunk 3]
```diff
@@ -6,12 +6,31 @@
...
+    public Dictionary<String, Object> bookDetails(@PathVariable(name = "id") Integer id) {
```
Comment (Szczepaniak-M) [inline src/main/java/pl/akai/bookcrossing/book/BookRestController.java]: Do poprawy - zamiast Dictionary Map: https://docs.oracle.com/javase/7/docs/api/java/util/Dictionary.html

**Review by Szczepaniak-M — CHANGES_REQUESTED** [review]

File: src/main/java/pl/akai/bookcrossing/book/BookRestController.java [hunk 4]
```diff
@@ -6,12 +6,31 @@
...
+        Dictionary<String, Object> dict = new Hashtable<>();
+        dict.put("bookDetails", bookBean.getBookById(id));
```
Comment (Szczepaniak-M) [inline src/main/java/pl/akai/bookcrossing/book/BookRestController.java]:
```suggestion
        dict.put("book", bookBean.getBookById(id));
```

**Review by Szczepaniak-M — CHANGES_REQUESTED** [review]

File: src/main/java/pl/akai/bookcrossing/book/BookRestController.java (hunk 3 above)
Comment (sniochup) [reply src/main/java/pl/akai/bookcrossing/book/BookRestController.java]: Poprawiono: https://github.com/akai-org/AKAI-bookcrossing/commit/e9937d04d4832dcececfbcfa79d6deca65b32f84

**Review by Szczepaniak-M — APPROVED** [review]

**Pull request merged**